src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -135,25 +162,171 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
+            var transactionIdsDescribeAclBindings = GetMatchingAllowAclBindings(adminClient, AclOperation.Describe).Where(x => x.Pattern.Type == ResourceType.Unknown);
+
+            var topicsWriteAclBindings = GetMatchingAllowAclBindings(adminClient, AclOperation.Write, ResourceType.Topic);
+
+            bool validTransactionWriteAcl =
+                transactionIdsWriteAclBindings.Any(transactionIdsAclBinding => AssertAclMatchingName(TransactionalId, transactionIdsAclBinding));
+            bool validTransactionDescribeAcl =
+                transactionIdsDescribeAclBindings.Any(transactionIdsAclBinding => AssertAclMatchingName(TransactionalId, transactionIdsAclBinding));
+            bool validTopicWriteAcl = topicsWriteAclBindings.Any(topicsAclBinding => AssertAclMatchingName(topicName, topicsAclBinding));
+
+            if (validTransactionWriteAcl && validTopicWriteAcl && validTransactionDescribeAcl)
             {
-                _producer.AbortTransaction();
+                return false;
             }
-            catch (Exception e)
+
+            var msgBuilder = new StringBuilder($"ETL process: {Name}. Failed to load data for the producer instance. " +
+                                               $"Please check your Kafka security configuration and ensure all of the (RESOURCE : OPERATION) ACL bindings listed below are set to ALLOW: {Environment.NewLine}");
+
+            if ((validTransactionDescribeAcl && validTransactionWriteAcl) == false)
             {
-                if (Logger.IsOperationsEnabled)
-                    Logger.Operations($" ETL process: {Name}. Aborting Kafka transaction failed.", e);
+                msgBuilder.Append($"Transactional ID '{TransactionalId}' : ");
+                if (validTransactionWriteAcl == false)
+                    msgBuilder.Append("WRITE operation ");
+                if (validTransactionDescribeAcl == false)
+                    msgBuilder.Append("DESCRIBE operation");
+                msgBuilder.Append(Environment.NewLine);
             }
 
-            throw new QueueLoadException(ex.Message, ex);
+            
+            if (validTopicWriteAcl == false)
+                msgBuilder.Append($"Topic '{topicName}' : WRITE operation");
+
+            message = msgBuilder.ToString();
+
+        }
+        catch (Exception)
+        {
+            return false;
         }
 
-        return count;
+        return true;
+    }
+
+    private static bool AssertAclMatchingName(string expectedName, AclBinding aclBinding)
+    {
+        return aclBinding.Pattern.ResourcePatternType switch
+        {
+            ResourcePatternType.Literal => aclBinding.Pattern.Name == expectedName,
+            ResourcePatternType.Prefixed => expectedName.StartsWith(aclBinding.Pattern.Name),
+            _ => throw new NotSupportedException()
```

Comment:
Also need to check the match type, right? Because currently it will throw.

Reply:
Please provide a message to the exception

Reply:
https://github.com/ravendb/ravendb/pull/17577/commits/6c7472d42b00f496f75e104e33c02da80a9aeabb

handling match in progress

Reply:
match also handled